src/controller/testController.ts

```diff
@@ -265,23 +265,57 @@ function enqueueTestMethods(testItems: TestItem[], run: TestRun): void {
  * @returns
  */
 async function getIncludedItems(request: TestRunRequest, token?: CancellationToken): Promise<TestItem[]> {
-   let testItems: TestItem[] = [];
-   if (request.include) {
-       testItems.push(...request.include);
-   } else {
-       testController?.items.forEach((item: TestItem) => {
-           testItems.push(item);
-       });
-   }
-   if (testItems.length === 0) {
-       return [];
-   }
-   removeTestInvocations(testItems);
-   testItems = await expandTests(testItems, TestLevel.Class, token);
-   // @ts-expect-error
-   const excludingItems: TestItem[] = await expandTests(request.exclude || [], TestLevel.Class, token);
-   testItems = _.differenceBy(testItems, excludingItems, 'id');
-   return testItems;
+    let testItems: TestItem[] = [];
+    if (request.include) {
+        testItems.push(...request.include);
+    } else {
+        testController?.items.forEach((item: TestItem) => {
+            testItems.push(item);
+        });
+    }
+    if (testItems.length === 0) {
+        return [];
+    }
+    if (!handleInvocationRerun(testItems)) { // a single invocation re-run will just update the item
+        removeTestInvocations(testItems);
+    }
+    testItems = await expandTests(testItems, TestLevel.Class, token);
+    // @ts-expect-error
+    const excludingItems: TestItem[] = await expandTests(request.exclude || [], TestLevel.Class, token);
+    testItems = _.differenceBy(testItems, excludingItems, 'id');
+    return testItems;
+}
+
+/**
+ * Check and preparation in case a single invocation of a parameterized test shall be re-run.
+ * @param testItems
+ * @returns true if a single invocation shall be re-run, otherwise false
+ */
+function handleInvocationRerun(testItems: TestItem[]): boolean {
```

Comment:
There is another issue might happen here. The current Testing explorer supports multi-selection. Users may select two invocation nodes and run them, which is not supported on our side now. Can we add one more check here? If there are more than 1 invocation nodes selected, we skip the test run and pop-up a message telling user that it's not supported?

Reply:
I added the check and an additional one to handle the case of selecting a mix of invocation and non-invocation items. The behavior there could be improved (e.g. checking whether the invocation is already included in the selected parts of the test-tree). However this may be fine for supporting the primary use-case of this feature?